### ISSUE TITLE
search crashes when given faulty regexp

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#3100]
I guess the side-effects of the #3089 recently merged
```shell
$> datalad search bids.subject.iq:.*            
[INFO   ]                           

$> datalad search bids.subject.iq:* 
[ERROR  ] nothing to repeat [re.py:_compile:251] (error) 

$> datalad --version
datalad 0.11.1.dev108
```

edit - empty message is the side effect -- crash is still there with 0.11.1 release
```shell
$> datalad search bids.subject.iq:*
[ERROR  ] nothing to repeat [re.py:_compile:251] (error) 

$> datalad search bids.subject.iq:.*

$> datalad --version
datalad 0.11.1
```

**Comment by kyleam** [comment]
The empty message should be the same as the later issue (#3195) and is fixed with #3222.

The compilation error happens here:

https://github.com/datalad/datalad/blob/930f464bdaaa154eebfc537d6e7fa40d379f1739/datalad/metadata/search.py#L880-L884

**Comment by yarikoptic** [comment]
I do not think there is anything else we should do about this  and error message says it all:

```shell
(git-annex)lena:~/datalad[master]
$> datalad search bids.subject.iq:.*
[INFO   ] no hits 

$> datalad search bids.subject.iq:* 
[ERROR  ] regular expression '(?i)*' (original: '*') is incorrect: nothing to repeat at position 4 [search.py:_compile_query:991] (ValueError) 

$> datalad --version
datalad 0.14.0.dev14
```
